js/common/bootstrap/modules/buffer.js

```diff
@@ -421,9 +413,15 @@ global.DEFINE_MODULE('buffer', (function () {
 
   Buffer.prototype.toJSON = function () {
     if (this.parent) {
-      return Array.prototype.slice.call(this.parent, this.offset, this.offset + this.length);
+      return {
+        type: "Buffer",
+        data: Array.prototype.slice.call(this.parent, this.offset, this.offset + this.length)
+      };
     }
-    return Array.prototype.slice.call(this, 0);
+    return {
+      type: "Buffer",
+      data: Array.prototype.slice.call(this, 0)
+    };
```

Comment:
This might be considered breaking but it's how Node buffers do. There's no good use case for converting buffers to JSON though. Not sure about this change.